parsing/lexer.mll

```diff
@@ -851,6 +851,14 @@ and directive already_consumed = parse
             directive_error lexbuf "line number out of range"
               ~already_consumed ~directive
         | line_num ->
+           (* If we're at the beginning of the file and claiming to be at the
+              beginning of a different file, then update [Location.input_name]
+              for better errors. *)
```

Comment:
There are two reasonable definitions of "we're at the beginning of the file":
  1. It's the first byte of the file
  2. Either (1), or a line directive has just reset the line number to 1.

Which is intended here?

Reply:
Right, sorry, I meant to leave a note on the PR about this.

The former seems preferable (at least, if this heuristic is reasonable at all, which I'm on the fence about!).  But the latter was more obvious to me how to implement and seemed fine for testing this on the tree, which is my next step.